vector-db-proxy/src/adaptors/pinecone/apis.rs

```diff
@@ -44,15 +44,7 @@ impl VectorDatabase for PineconeClient {
             collection_name: search_request.collection.clone(),
             collection_metadata: None,
         };
-        let region = search_request.region.unwrap_or_default();
-        let namespace = search_request
-            .clone()
-            .namespace
-            .map_or(search_request.clone().collection, |n| n);
-        let index_name = search_request
-            .byo_vector_db
-            .map_or(Region::to_str(region), |_k| namespace.as_str())
-            .to_string();
+        let index_name = search_request.clone().collection;
```

Comment:
In pinecone because there is a limit of 20 indexes when using serverless, we allocate the collection name to a namespace rather than an index. So by setting the collection name as the index name as you have done here this suggests that there's a index per collection name, which is not what we want

Reply:
Yup thats 20 indexes in the standard plan and each index can have 10,000 namespaces.
The flow I pictured so far
1. User selects which index to use when creating a datasource
2. Vector db proxy check if the index exist, if not it creates the index
3. Vector db proxy upserts the embeddings to the index to namespace where namespace is equal to the datasource

Reply:
![image](https://github.com/user-attachments/assets/e2c86026-b488-43e5-b6ee-1e786de810ce)

Reply:
When you say "users select an index" select from where?